src/main/java/ua/danit/final_project/entities/Permission.java

```diff
@@ -0,0 +1,39 @@
+package ua.danit.final_project.entities;
+
+import lombok.Data;
+import javax.persistence.Id;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+import javax.persistence.Table;
+import java.io.Serializable;
+import java.util.List;
+
+@Entity
+@Table(name = "permission")
+@Data
+public class Permission implements Serializable {
+
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  private Long id;
+
+  @Column(name = "p_name", nullable = false, unique = true)
+  private String name;
+
+  @ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
+  @JoinTable(
+          name = "role_permission",
+          joinColumns = {@JoinColumn(name = "p_id")},
+          inverseJoinColumns = {@JoinColumn(name = "r_id")})
+  public List<Role> roles;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Make "roles" private or transient. ([squid:S1948](https://sonarcloud.io/organizations/final-project/rules#rule_key=squid%3AS1948))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=final-project&pullRequest=82&issues=AWVYzZe12xJdnw6Qj_eS&open=AWVYzZe12xJdnw6Qj_eS)

[ISSUE_KEY]: AWVYzZe12xJdnw6Qj_eS